### PR TITLE
Adds "angular-momenta" to Calculator class _getter dictionary.

### PR DESCRIPTION
**Pull request by kvkarandashev**
Created a function for exporting angular momenta of calculator's atomic orbitals, as discussed in #49

### CODE REVIEW

**Review by awvwgk — CHANGES_REQUESTED** [review]
Preferably, we can return the shell angular momenta rather than the orbital ones, using the `orbital-map` those can be easily expanded and don't require to produce redundant information in the API.

Also, please add tests for all your changes.

**Comment by kvkarandashev** [comment]
Thank you for the feedback! I decided to make API's new output part of the test_gfn2 procedure in test_interface, but I am afraid I have trouble passing this test both with my branch and tblite/tblite main branch. On closer inspection, for tblite/tblite main branch I can pass all tests in tblite.test_interface but for test_gfn2 and test_gfn2_mindless; I also checked that running "meson test -C _build --print-errorlogs" passes all tests but for "xtb-param" and "xtb-external", for which it faces a timeout.
Could you check whether the issue is with the tests or with my installation?

**Comment by awvwgk** [comment]
>  Could you check whether the issue is with the tests or with my installation?

Depending on your machine setup, it might be worth to limit the number of threads and/or concurrent runs to avoid over-subscription of the physical cores:

```sh
export OMP_NUM_THREADS=$(nproc),1
meson test -C _build --print-errorlogs --num-processes 1
```

**Comment by kvkarandashev** [comment]
Thank you! Running "meson test" with only one process and one OMP thread did solve the timeout issue; I now pass 33 tests with 1 expected failure ("tblite:app / noargs").

However, I am afraid that the problem with reproducing "test_interface.test_gfn2" and "test_interface.test_gfn2_mindless" persists as I fail "assert res.get("energy") == ..." in both cases despite passing "tblite:unit / gfn2-xtb" of the meson test.

**Comment by awvwgk** [comment]
> However, I am afraid that the problem with reproducing "test_interface.test_gfn2" and "test_interface.test_gfn2_mindless" persists as I fail "assert res.get("energy") == ..." in both cases despite passing "tblite:unit / gfn2-xtb" of the meson test.

Strange indeed, the setup for the continuous integration passes the Python API with your modification, so it should be fine I guess.

**Comment by awvwgk** [comment]
Would `angular_momenta` be sufficient here to keep the name short?

**Comment by kvkarandashev** [comment]
I have changed python API to return angular momenta corresponding to shells rather than AO's and added the corresponding test to test_gfn2 in tblite.test_interface. Unfortunately, I still have trouble passing this test due to energy assertion error, but the latter is raised after the angular momenta value assertion is passed successfully.
I will revisit the tests in test_interface sometime this week and if the problem is still there and I reproduce it on a different machine I will create an issue for it.

**Comment by kvkarandashev** [comment]
> Would `angular_momenta` be sufficient here to keep the name short?

Now that I think about it yes, it is more fitting than "shell_angular_momenta". I have made the corresponding commit.

File: include/tblite/calculator.h [hunk 1]
```diff
@@ -113,6 +113,12 @@ tblite_get_calculator_shell_map(tblite_context /* ctx */,
                             tblite_calculator /* calc */,
                             int* /* sh2at */);
 
+/// Query calculator for angular momenta of shells
+TBLITE_API_ENTRY void TBLITE_API_CALL
+tblite_get_calculator_angular_momenta(tblite_context /* ctx */,
+                              tblite_calculator /* calc */,
+                              int* /* am */);
```
Comment (awvwgk) [inline include/tblite/calculator.h]: Hanging indentation for C:

```suggestion
tblite_get_calculator_angular_momenta(tblite_context /* ctx */,
                                      tblite_calculator /* calc */,
                                      int* /* am */);
```

**Review by awvwgk — CHANGES_REQUESTED** [review]
Please also add tests for the C-API in [`test/api/main.c`](https://github.com/tblite/tblite/blob/main/test/api/main.c).

Especially the failing case at

https://github.com/tblite/tblite/blob/4a82f966e919311e7cb489f26e298df5592e6bc5/test/api/main.c#L254

File: src/tblite/api/calculator.f90 [hunk 2]
```diff
@@ -401,6 +401,35 @@ subroutine get_calculator_shell_map(vctx, vcalc, imap) &
...
 
+subroutine get_calculator_angular_momenta(vctx, vcalc, am) &
+      & bind(C, name=namespace//"get_calculator_angular_momenta")
+   type(c_ptr), value :: vctx
+   type(vp_context), pointer :: ctx
+   type(c_ptr), value :: vcalc
+   type(vp_calculator), pointer :: calc
+   integer(c_int), intent(inout) :: am(*)
+   type(error_type), allocatable :: error
+   integer:: ish
```
Comment (awvwgk) [inline src/tblite/api/calculator.f90]:
```suggestion
   integer :: ish
```

File: src/tblite/api/calculator.f90 [hunk 3]
```diff
@@ -401,6 +401,35 @@ subroutine get_calculator_shell_map(vctx, vcalc, imap) &
...
+      call fatal_error(error, "Calculator object is missing")
+      call ctx%ptr%set_error(error)
+      return
+   end if
+   call c_f_pointer(vcalc, calc)
+
+   associate(bas => calc%ptr%bas)
+      do ish=1, bas%nsh
+         am(ish)=(bas%nao_sh(ish)-1)/2
+      enddo
```
Comment (awvwgk) [inline src/tblite/api/calculator.f90]: That's a clever way to retrieve the angular momentum of a shell, clever implementations deserve a comment explaining why they were done this way to spare future generations the mental exercise to figure out what is going on here.
```suggestion
      do ish = 1, bas%nsh
         ! The angular momentum is only available for each species,
         ! however the species to atom mapping is unavailable in this API call.
         ! Therefore, we calculate it by using 2*l + 1 == nao
         am(ish) = (bas%nao_sh(ish)-1)/2
      end do
```

**Review by awvwgk — COMMENTED** [review]
Some final style comments

File: test/api/main.c [hunk 4]
```diff
@@ -971,6 +987,31 @@ test_gfn2_si5h12 (void)
...
+
+   tblite_get_calculator_shell_map(ctx, calc, sh2at);
+   if (tblite_check_context(ctx)) goto err;
+
+   if (!check(sum_int(nshells, sh2at), 156, "dimension error")) goto err;
+
+   tblite_get_calculator_angular_momenta(ctx, calc, am);
+   if (tblite_check_context(ctx)) goto err;
+
+   if (!check(sum_int(nshells, am), 15, "dimension error")) goto err;
```
Comment (awvwgk) [inline test/api/main.c]: Summing the angular momenta doesn't make sense, better check for a specific atom, like a hydrogen where the momenta are clear. Also the error message is incorrect in this context.

**Comment by kvkarandashev** [comment]
OK, would it be even better to compare the integer array to a benchmark array entry-wise, just like in test_interface.test_gfn2? Or would it be too much data stored for the test?

**Review by awvwgk — APPROVED** [review]
Thanks for sharing, looks good to me now.

**Comment by kvkarandashev** [comment]
Thanks a lot!

**Pull request merged**